nutils/function.py

```diff
@@ -1987,7 +1987,7 @@ def norm2(__arg: IntoArray, axis: Union[int, Sequence[int]] = -1) -> Array:
   '''
 
   arg = Array.cast(__arg)
-  return sqrt(sum(multiply(arg, arg), axis))
+  return sqrt(dot(arg, arg, axis))
```

Comment:
change to vdot

Reply:
Since `numpy.vdot` always returns a scalar, I've changed this to `sqrt(sum(multiply(arg, conjugate(arg)), axis))`.